File: python/2.7/factory/Pizza.py
from abc import ABCMeta, abstractmethod

class Pizza:
    __metaclass__ = ABCMeta

    def __init__(self):
        self.name = None

        self.dough = None
        self.sauce = None
        self.veggies = None
        self.cheese = None
        self.pepperoni = None
        self.clam = None

    @abstractmethod
    def prepare(self):
        pass

    def bake(self):
        print "Bake for 25 minutes at 350"

    def cut(self):
        print "Cutting the pizza into diagonal slices"

    def box(self):
        print "Place pizza in official PizzaStore box"

    def setName(self, name):
        self.name = name

    def getName(self):
        return self.name

    def __str__(self):
        result = []
        result.append("---- %s ----\n" %(self.name))
        if self.dough != None:
            result.append(self.dough)
            result.append("\n")
        if self.sauce != None:
            result.append(self.sauce)
            result.append("\n")
        if self.cheese != None:
            result.append(self.cheese)
            result.append("\n")
        if self.veggies != None:
            for veggie in self.veggies:
                result.append(veggie)
                result.append(", ")
            result.pop()
        if self.clam != None:
            result.append(self.clam)
            result.append("\n")
        if self.pepperoni != None:
            result.append(self.pepperoni)
            result.append("\n")
        return "".join(result)

File: js/node/factory/com/pizza/pizza/Pizza.js
Pizza = function() {
    this.name = null;

    this.dough = null;
    this.sauce = null;
    this.veggies = null;
    this.cheese = null;
    this.pepperoni = null;
    this.clam = null;
}
Pizza.prototype.prepare = function() {
    throw new Error("This method must be overwritten!");
}
Pizza.prototype.bake = function() {
    console.log("Bake for 25 minutes at 350");
}
Pizza.prototype.cut = function() {
    console.log("Cutting the pizza into diagonal slices");
}
Pizza.prototype.box = function() {
    console.log("Place pizza in official PizzaStore box");
}
Pizza.prototype.setName = function(name) {
    this.name = name;
}
Pizza.prototype.getName = function() {
    return this.name;
}
Pizza.prototype.toString = function() {
    var result = [];
    result.push("---- " + this.name + " ----\n");
    if(this.dough != null) {
        result.push(this.dough);
        result.push("\n");
    }
    if(this.sauce != null) {
        result.push(this.sauce);
        result.push("\n");
    }
    if(this.cheese != null) {
        result.push(this.cheese);
        result.push("\n");
    }
    if(this.veggies != null) {
        for(var veggie in this.veggies) {
            result.push(this.veggies[veggie]);
            result.push(", ");
        }
        result.pop();
        result.push("\n");
    }
    if(this.clam != null) {
        result.push(this.clam);
        result.push("\n");
    }
    if(this.pepperoni != null) {
        result.push(this.pepperoni);
        result.push("\n");
    }
    return result.join("");
}